### PR TITLE
4L/640d width: +25% model capacity cold-start for wall-shear improvement

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Increasing hidden dimension from 512 to 640 with the full SOTA stack (learnable-PE + Lion + grad-ema + beta-NLL + k1_k2 curvature) cold-start will improve model expressiveness for τ_y/τ_z wall-shear prediction without the catastrophic undertraining seen in the 768d attempts (PRs #639, #659). The key constraint is slices: 640/128=5.0, a clean integer ratio, so no slice padding is needed. Prior attempts at 768d failed not because of architecture but because the parameter increase required more training epochs than the timeout allowed — 640d is a moderate +25% width step that balances expressiveness with trainability.

## Instructions

Run a cold-start from scratch with model-hidden-dim=640, model-heads=10 (640/64=10 heads, maintaining d_head=64). All other hyperparameters identical to the yi SOTA cold-start. Use 4 GPUs.

**Reproduce command:**
```bash
cd target/
python train.py --help  # verify flags
CUDA_VISIBLE_DEVICES=0,1,2,3 torchrun --standalone --nproc_per_node=4 train.py \
  --agent dazai \
  --wandb-group yi-round38-width-640d \
  --wandb-name "dazai/width-640d-cold-start" \
  --learnable-pe \
  --optimizer lion \
  --lr 1e-4 \
  --weight-decay 5e-4 \
  --clip-grad-norm 0.5 \
  --grad-ema-alpha 0.5 \
  --ema-decay 0.999 \
  --model-layers 4 \
  --model-hidden-dim 640 \
  --model-heads 10 \
  --model-slices 128 \
  --batch-size 4 \
  --beta-nll-beta 0.5 \
  --surface-curvature-features k1_k2 \
  --train-surface-points 65536 \
  --eval-surface-points 65536 \
  --train-volume-points 65536 \
  --eval-volume-points 65536 \
  --validation-every 1 \
  --no-compile-model \
  --lr-warmup-epochs 1 \
  --gradient-log-every 100 \
  --weight-log-every 100
```

Note: `--model-hidden-dim 640 --model-heads 10` gives d_head=64 (same as 512/8). Slices=128 divides 640 cleanly: 640/128=5. Check VRAM fits in 96GB before long runs; if OOM reduce to --batch-size 2.

Run until SENPAI_TIMEOUT_MINUTES. Report val_abupt and all 7 sub-metrics at the best validation checkpoint. Monitor early epoch progress carefully — if val_abupt at epoch 2 is worse than 15% (sign of slow start due to extra parameters), note it and continue.

## Baseline

Current yi SOTA: **PR #658** (nezuko, EMA best-ckpt from lr=5e-6 SWA run), W&B run `pxsnrw36`

| Metric | val | test |
|---|---:|---:|
| abupt_axis_mean_rel_l2_pct | **7.3914%** | 8.7189% |
| surface_pressure_rel_l2_pct | 4.8552% | — |
| wall_shear_rel_l2_pct | 8.3192% | — |
| volume_pressure_rel_l2_pct | 4.3156% | ~11.46% |
| wall_shear_x_rel_l2_pct (τ_x) | 7.1166% | — |
| wall_shear_y_rel_l2_pct (τ_y) | 9.6123% | 9.7675% |
| wall_shear_z_rel_l2_pct (τ_z) | 11.0573% | 10.8851% |

**AB-UPT reference targets (test):** surface_pressure 3.82%, wall_shear 7.29%, volume_pressure 6.08%, τ_x 5.35%, τ_y 3.65%, τ_z 3.63%

To beat: val_abupt < 7.3914%

**Context on 768d failures:** PR #639 val_abupt=12.9% (undertraining), PR #659 val_abupt=8.6% (improved but still not SOTA). 640d is expected to converge faster than 768d due to fewer parameters.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Kicking off your experiment. A few important corrections before you start:

**Merge bar correction**: The baseline in the PR body is stale. PR #681 (nezuko, terminal LR polish lr=3e-7) has since been merged and is the new SOTA. The correct target is **val_abupt < 7.3767%** (not 7.3914%). W&B run `dc031qpt`. Updated full baseline:

| Metric | val | test |
|---|---:|---:|
| abupt_axis_mean_rel_l2_pct | **7.3767%** | **8.7015%** |
| surface_pressure_rel_l2_pct | 4.8515% | 4.6236% |
| wall_shear_rel_l2_pct | 8.3016% | 8.3214% |
| volume_pressure_rel_l2_pct | 4.3066% | 11.3738% |
| wall_shear_x_rel_l2_pct (τ_x) | 7.1045% | 7.1753% |
| wall_shear_y_rel_l2_pct (τ_y) | 9.5832% | 9.5964% |
| wall_shear_z_rel_l2_pct (τ_z) | 11.0377% | 10.7383% |

**Kill-threshold correction**: Use `val_abupt < 50` (less-than), NOT `>= 50`. The correct gate is: kill the run if val_abupt is **not** below 50% at EP1. A healthy cold-start at EP1 should be 15–30%; if it's above 50% the run has diverged.

**4-GPU execution**: Run sequentially on all 4 GPUs as specified (`--nproc_per_node=4`). Do not run on fewer GPUs.

**EP1 gate**: After epoch 1 completes, report val_abupt. Expected healthy range for 640d cold-start at EP1: ~18–28% (the extra parameters slow early convergence slightly vs 512d but it should still hit <30%). If EP1 > 50%, kill and check configuration.

**VRAM check**: 640d with batch=4 should fit in 96GB comfortably. The hidden_dim=640/heads=10 gives d_head=64 identical to the SOTA. Slices=128 divides cleanly into 640 (640/128=5). If you do hit OOM, reduce to --batch-size 2 and note it.

**What to watch**: The key question is whether the +25% parameter increase provides enough extra capacity to close the τ_y/τ_z gap without requiring more epochs than the timeout allows. Report all 7 sub-metrics at each validation. If val_abupt at EP2 is still >15%, flag it but continue.

Please start the run and post an update after EP1. Good luck!

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #684 because ADVISOR: Closing orphaned draft. Student dazai is no longer in the active 16-student lineup; this draft assignment was never started..
